app/src/main/res/layout/activity_form.xml

```diff
@@ -0,0 +1,112 @@
+
+        app:layout_constraintStart_toStartOf="@+id/EditTextProjectName"
+        app:layout_constraintTop_toBottomOf="@+id/EditTextProjectName"
+        android:autofillHints=""></EditText>
+
+    <EditText
+        android:id="@+id/EditTextProjectName"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="46dp"
+        android:layout_marginTop="24dp"
+        android:hint="@string/project_name"
+        android:inputType="textPersonName"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/title_form"
+        android:autofillHints=""></EditText>
+
+    <EditText
+        android:id="@+id/EditTextTeacher"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="8dp"
+        android:hint="@string/teacher"
+        android:inputType="textPersonName"
+        app:layout_constraintStart_toStartOf="@+id/EditTextLaboratory"
+        app:layout_constraintTop_toBottomOf="@+id/EditTextLaboratory"
+        android:autofillHints=""></EditText>
+
+    <EditText
+        android:id="@+id/EditTextProjectTA"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="7dp"
+        android:hint="@string/project_TA"
+
+        android:inputType="textPersonName"
+        app:layout_constraintStart_toStartOf="@+id/EditTextTeacher"
+        app:layout_constraintTop_toBottomOf="@+id/EditTextTeacher"
+        android:autofillHints=""></EditText>
+
+    <EditText
+        android:id="@+id/nbOfParticipant"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:hint="@string/numberOfParticipant"
+        android:inputType="number"
+        app:layout_constraintStart_toStartOf="@+id/EditTextProjectTA"
+        app:layout_constraintTop_toBottomOf="@+id/EditTextProjectTA"
+        android:autofillHints=""></EditText>
+
+    <CheckBox
+        android:id="@+id/CheckBoxSP"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="9dp"
+        android:text="@string/semester_project"
+        app:layout_constraintEnd_toStartOf="@+id/CheckBoxMP"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/nbOfParticipant"></CheckBox>
+
+    <CheckBox
+        android:id="@+id/CheckBoxMP"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginEnd="29dp"
+        android:text="@string/master_project"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toEndOf="@+id/CheckBoxSP"
+        app:layout_constraintTop_toBottomOf="@+id/nbOfParticipant"></CheckBox>
+
+    <EditText
+        android:id="@+id/projectDescription"

```

Comment:
Because multiples elements will have a "projectDescription" for example, a better naming would be to contains the activity name (or at least to be uniquely identifiable).